File: backend/src/main/java/org/cms/Enums/Status.java
package org.cms.Enums;

import java.util.HashMap;
import java.util.Map;

public enum Status {
    ACTIVE(0),
    INACTIVE(1);

    private final int value;
    private static final Map<Integer, Status> map = new HashMap<>();

    private Status(int value) {
        this.value = value;
    }

    static {
        for(Status status : Status.values())
            map.put(status.value, status);
    }

    public static Status valueOf(int value) {
        return map.get(value);
    }

    private int getValue() {
        return value;
    }
}

File: frontend/src/enums/Status.ts
export enum Status {
    ACTIVE = 0,
    INACTIVE = 1
};

export namespace Status {
    export function valueOf(value: string | number): Status | ((value: string | number) => Status | typeof valueOf | undefined) | undefined {
        if (typeof value === 'string'){
            if(!(value in Status)) return undefined;
    
            return Status[value as keyof typeof Status];
        }

        if(typeof value === 'number'){
            return Object.values(Status).includes(value as Status) ? value as Status : undefined;
        }
    }
};